feat: aztec-up installs without docker

charlielye opened the pull request:
Now we're shipping `bb` in bb.js for supported architectures/os's, we can support installs without needing docker.
The user does however need a distribution with:
* The minimum required node version.
* Tools e.g: `make`, `gcc`, `python`, `git`.

_Note: You will not be able to do backend proving as we don't ship `bb-avm`. This is contract development only. Provers/validators are expected to still use the release image._

This PR:
* `boxes` and `aztec-up` no longer depends on `release-image`.
* We test `aztec-up` with a local npm registry (`verdaccio`) that we prime with the monorepo packages.
* New install script that installs correct versions of:
  * `nargo`
  * foundry
  * jq
  * packages `@aztec/aztec`, `@aztec/cli-wallet`, `bb` via `@aztec/bb.js`.
* `./bootstrap.sh versions` outputs all expected versions of relevant tools (used by aztec-up).
* `boxes` now build and test without any requirement on a release image (they start the local network directly).
* Remove unwanted `npx aztec-create` stuff from boxes.
* Remove some never used log commands from ci.sh.
* Rework some bootstrap release stuff so we can easily get paths to packages to be released (helps load into verdaccio).
* Remove interactive query from noir precommit hook as they can't be interactive.
* Move the nasty aztec cli extension script(s) out of `aztec-up` and into `yarn-project/aztec/scripts/aztec.sh` and make it the entrypoint. The script was only every meant to abstract away the docker container but additional functionality made its way in. This functionality should probably all be TS code, but for now at least it's owned by the cli package. Extensions are `test`, `compile`, `new`, `init`.

olehmisar commented:
hi. Any chance we can have automatic aztec version selection by specifying the aztec version inside a dotfile per each project. Similar to gaztec: https://github.com/nemi-fi/gaztec

charlielye commented:
* **#19261** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19261?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#18665** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18665?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/18665?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

ludamad reviewed (approved):
very cool. Do we at some point just release a contract dev image thats just the result of running aztec-up on a fresh ubuntu, and then strip everything contract dev-related out of release-image?

ludamad commented:
Thanks for doing this maddeningly detail-oriented work

charlielye commented:
> very cool. Do we at some point just release a contract dev image thats just the result of running aztec-up on a fresh ubuntu, and then strip everything contract dev-related out of release-image?

It's possible, but I also thought we'd probably just stop providing any kind of "contract dev" image, and just provide an image for node operators (of which the build of the image could possibly use the installer if we make bb-avm part of an npm package).

parts of the contract dev experience could work ok in the container (e.g running the TXE and local network), but things like the wallet, that wanted to use ssh agent were a right pain. Seems easier at this point to just commit to supporting all native installs, and just provide an image for infrastructure convienience.

charlielye committed:
feat: aztec-up installs without docker

Now we're shipping `bb` in bb.js for supported architectures/os's, we can support installs without needing docker.
The user does however need a distribution with:
* The minimum required node version.
* Tools e.g: `make`, `gcc`, `python`, `git`.

_Note: You will not be able to do backend proving as we don't ship `bb-avm`. This is contract development only. Provers/validators are expected to still use the release image._

This PR:
* `boxes` and `aztec-up` no longer depends on `release-image`.
* We test `aztec-up` with a local npm registry (`verdaccio`) that we prime with the monorepo packages.
* New install script that installs correct versions of:
  * `nargo`
  * foundry
  * jq
  * packages `@aztec/aztec`, `@aztec/cli-wallet`, `bb` via `@aztec/bb.js`.
* `./bootstrap.sh versions` outputs all expected versions of relevant tools (used by aztec-up).
* `boxes` now build and test without any requirement on a release image (they start the local network directly).
* Remove unwanted `npx aztec-create` stuff from boxes.
* Remove some never used log commands from ci.sh.
* Rework some bootstrap release stuff so we can easily get paths to packages to be released (helps load into verdaccio).
* Remove interactive query from noir precommit hook as they can't be interactive.
* Move the nasty aztec cli extension script(s) out of `aztec-up` and into `yarn-project/aztec/scripts/aztec.sh` and make it the entrypoint. The script was only every meant to abstract away the docker container but additional functionality made its way in. This functionality should probably all be TS code, but for now at least it's owned by the cli package. Extensions are `test`, `compile`, `new`, `init`.

Co-authored-by: iAmMichaelConnor <mike@aztecprotocol.com>
Co-authored-by: ludamad <domuradical@gmail.com>
